src/components/OnboardingManager.tsx

```diff
@@ -30,14 +33,17 @@ export default function OnboardingManager({
     // out of bounds redirect
     if (
       onboarding.progress < 0 ||
-      onboarding.progress >= onboarding.flow.length
+      onboarding.progress >= onboarding.flow.length ||
+      !profile?.userId ||
+      (profile?.profileReady && profile?.profileData)
     ) {
-      router.push('/cases');
+      router.push(CONFIG.homepage);
+      return;
     }
 
     if (pageProgress > onboarding.progress)
       router.push(`/onboarding/${onboarding.flow[onboarding.progress].url}`);
-  }, [onboarding, router, pageProgress]);
+  }, [onboarding, profile, router, pageProgress]);
```

Comment:
just to clarify, why is it necessary to have profile in the dependency list?
it is b/c the profile will change as the user goes through onboarding?

Reply:
Yes.

Reply:
ahhh ic, that makes sense; thank you!